packages/eslint-plugin/src/rules/no-type-imports-with-verbatim-side-effects.ts

```diff
@@ -0,0 +1,78 @@
+import type { TSESLint, TSESTree } from '@typescript-eslint/utils';
+import { AST_NODE_TYPES } from '@typescript-eslint/utils';
+
+import * as util from '../util';
+
+type Options = [];
+type MessageIds = 'useTopLevelQualifier';
+
+export default util.createRule<Options, MessageIds>({
+  name: 'no-type-imports-with-verbatim-side-effects',
+  meta: {
+    type: 'problem',
+    docs: {
+      description:
+        'Enforce the use of top-level import type qualifier when an import only has specifiers with inline type qualifiers',
+      recommended: false,
+    },
+    fixable: 'code',
+    messages: {
+      useTopLevelQualifier:
+        'TypeScript will only remove the inline type specifiers which will leave behind a side effect import at runtime. Convert this to a top-level type qualifier to properly remove the entire import.',
+    },
+    schema: [],
+  },
+  defaultOptions: [],
+  create(context) {
+    const sourceCode = context.getSourceCode();
+    return {
+      ImportDeclaration(node): void {
+        if (node.importKind === 'type') {
+          return;
+        }
+
+        const specifiers: TSESTree.ImportSpecifier[] = [];
+        for (const specifier of node.specifiers) {
+          if (specifier.type !== AST_NODE_TYPES.ImportSpecifier) {
+            return;
+          }
+          if (specifier.importKind !== 'type') {
+            return;
+          }
```

Comment:
[Nit] Not blocking at all, just a thought to save a couple of lines...

```suggestion
          if (
            specifier.type !== AST_NODE_TYPES.ImportSpecifier ||
            specifier.importKind !== 'type'
          ) {
```